src/modules/Core.py

```diff
@@ -68,6 +69,32 @@ def get_parameter_options(self) -> dict:
         raise NotImplementedError("Please don't use the base version of get_parameter_options. "
                                   "Implement your own override instead.")
 
+    def get_available_submodules(self, option: list) -> list:
+        """
+        If the module has submodules depending on certain options, this method should adjust the submodule_options
+        accordingly.
+
+        :param option: List of chosen options
+        :type option: list
+        :return: List of available submodules
+        :rtype: list
+        """
+        return []
+
+    def get_depending_parameters(self, option: str, config: dict) -> dict:
+        """
+        If the module has parameters depending on certain options, this method should return the parameters for the
+        given option.
+
+        :param option: The chosen option
+        :type option: str
```

Comment:
@Marvmann The specification of types is no longer needed since #142 right? This applies to multiple places in this PR.

Reply:
Correct. I will make sure the style matches the rest of the code after the changes in #142. Same for syncing it with the new structure introduced in Release 2.1.1.